src/tablecloth/time/api/slice.clj

```diff
@@ -66,28 +73,35 @@
   | 1973 |  3 |
   "
   ([dataset from to] (slice dataset from to nil))
-  ([dataset from to options]
+  ([dataset from to {:keys [result-type]
+                     :or {result-type :as-dataset}}]
    (let [build-err-msg (fn [^java.lang.Exception err arg-symbol time-unit]
                          (let [msg-str "Unable to parse `%s` date string. Its format may not match the expected format for the index time unit: %s. "]
                            (str (format msg-str arg-symbol time-unit) (.getMessage err))))
-         time-unit (get-index-type dataset)
+         time-unit (if (can-identify-index-column? dataset)
+                     (unpack-datatype (index-column-datatype dataset))
+                     (throw (Exception. "Unable to auto detect time column to serve as index. Please specify the index using `index-by`.")))
          from-key (cond
                     (or (int? from)
-                        (instance? java.time.temporal.Temporal from)) from
+                        (time-datatype? (get-datatype from))) from
                     :else (try
-                            (parse-datetime-str time-unit from)
+                            (parse-datetime-str (unpack-datatype time-unit) from)
```

Comment:
Note the `unpack-datatype` here. This will change `:packed-local-date` to `:local-date`, or pass the type through if it is not a packed type.